Support for Netezza Performance Server as a data source

aniket-s-kulkarni opened the pull request:
## What type of PR is this? 
<!-- Check all that apply, delete what doesn't apply. -->

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] New Query Runner (Data Source) 
- [ ] New Alert Destination
- [ ] Other

## Description
Adding a query runner to support Netezza Performance Server as a data source. 

## How is this tested?

- [ ] Unit tests (pytest, jest)
- [ ] E2E Tests (Cypress)
- [x] Manually
- [ ] N/A

Manual testing included 
- connecting a variety of Netezza backends
- running individual queries
- ensuring that schema discovery is working well

## Related Tickets & Documents
<!-- If applicable, please include a link to your documentation PR against getredash/website -->

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
<img width="1080" alt="New data source" src="https://user-images.githubusercontent.com/7376200/174418350-29407034-8e55-4fbc-9246-3bd3dc5abcbd.png">

<img width="908" alt="New data source " src="https://user-images.githubusercontent.com/7376200/174418379-12bcfe25-a5b6-45ed-9373-af600962fb2f.png">

aniket-s-kulkarni commented:
I am confused by the failing check about UI review. Do I need to add a different label for the right review team to take a look at this?

susodapop commented on requirements.txt:
```diff
@@ -66,4 +66,6 @@ werkzeug==0.16.1
-advocate==1.0.0
\ No newline at end of file
+advocate==1.0.0
+nzpy>=1.15
+nzalchemy
```
Query runner dependencies belong in `requirements_all_ds.txt`. `requirements.txt` is only for Redash's own dependencies.

You can see [this article](https://redash.io/help/open-source/dev-guide/write-a-query-runner#Finishing-up) for more info.

```suggestion
```

susodapop commented on requirements_all_ds.txt:
```diff
@@ -42,3 +42,4 @@ xlrd==2.0.1
 openpyxl==3.0.7
 firebolt-sdk
 pandas==1.3.4
+nzpy
```
Mirrors what was added to `requirements.txt` originally.

```suggestion
nzpy>=1.15
nzalchemy
```

susodapop commented on requirements_dev.txt:
```diff
@@ -12,3 +12,5 @@ ptvsd==4.3.2
 freezegun==0.3.12
 watchdog==0.9.0
 ptpython==3.0.17
+nzpy>=1.15
+nzalchemy
```
`requirements_dev.txt` is only for development dependencies, needed for unit tests and other general development tasks. See earlier comment about where to specify query runner dependencies.

```suggestion
```

susodapop reviewed (changes requested):
Thanks so much for this contribution. 

I requested a two small changes to avoid build issues.

I haven't tested this against an active Netezza system. But we'll take your word for it that this succeeds. Happy to merge once the requirements files are fixed.

susodapop commented:
Our percy check is flaky right now. I've manually verified there are no UI changes that would block merging this PR.

aniket-s-kulkarni commented:
> Our percy check is flaky right now. I've manually verified there are no UI changes that would block merging this PR.



> Thanks so much for this contribution.
> 
> I requested a two small changes to avoid build issues.
> 
> I haven't tested this against an active Netezza system. But we'll take your word for it that this succeeds. Happy to merge once the requirements files are fixed.

Thanks @susodapop

I have addressed the review comments and 

> Thanks so much for this contribution.
> 
> I requested a two small changes to avoid build issues.
> 
> I haven't tested this against an active Netezza system. But we'll take your word for it that this succeeds. Happy to merge once the requirements files are fixed.


@susodapop thanks, yes I have addressed the review comments and resolved the conflicts. I do have access to Netezza and therefore have been able to test it out.

susodapop commented:
Thank you! It's nighttime GMT-5. I'll review tomorrow and follow-up with you. Huge thanks for your responsiveness! 🙏

susodapop reviewed (approved):
LGTM
